### PR TITLE
Creates and crons (daily) the retention_stats cron.

### PR DESCRIPTION
**Pull request by ashercodeorg**
This data will be used in upcoming PRs to surface retention information internally.

### CODE REVIEW

File: bin/cron/retention_stats [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  drop_off = UserScript.
+    where(last_progress_at: (Time.now - 4 * ONE_WEEK)..(Time.now - 2 * ONE_WEEK))
+  script_starts = drop_off.group(:script_id).count
+  script_finishes = drop_off.where.not(completed_at: nil).group(:script_id).count
+  # 
+  drop_off_with_levels = drop_off.
+    joins("INNER JOIN user_levels ON user_levels.user_id = user_scripts.user_id AND user_levels.script_id = user_scripts.script_id")
+  # Determine, for each (user, script), the number of levels the user attempted. Then construct a
+  # hash of hashes of the form {script_id: {level_count: number_of_users}}.
+  levels = Hash.new {|hash, key| hash[key] = Hash.new {|hash, key| hash[key] = 0}}
```
Comment (Hamms) [inline bin/cron/retention_stats]: I think you should be able to just do `levels = Hash.new {|hash, key| hash[key] = Hash.new(0) }`;

``` ruby
$ irb
@irb(main):001:0> levels = Hash.new {|hash, key| hash[key] = Hash.new(0) }
=> {}
@irb(main):002:0> levels[:foo][0] += 1
=> 1
@irb(main):003:0> levels[:foo][0]
=> 1
@irb(main):004:0> levels[:foo][1]
=> 0
```

File: bin/cron/retention_stats [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  drop_off_with_levels.
+    group('user_levels.user_id', 'user_levels.script_id').
+    count.
+    each{|pair, count| levels[pair.second][count] += 1}
+  # Determine, for each (user, script, stage), the number of levels attempted. The construct a hash
```
Comment (Hamms) [inline bin/cron/retention_stats]: nit: "Then construct"

**Comment by Hamms** [comment]
LGTM!

File: bin/cron/retention_stats (hunk 1 above)
Comment (ashercodeorg) [inline bin/cron/retention_stats]: Done.

File: bin/cron/retention_stats (hunk 2 above)
Comment (ashercodeorg) [inline bin/cron/retention_stats]: Done.

**Pull request merged**